python/lsst/ap/association/dia_collection.py

```diff
@@ -188,15 +232,51 @@ def match(self, dia_source_catalog, scores):
         dia_source_catalog : an lsst.afw.SourceCatalog
             A contiguous catalog of dia_sources for which the set of scores
             has been computed on with DIAObjectCollection.score.
-        scores : (N DIAObject by M DIASource) float array
-            A ndarray containing the scores between each DIAObject and
-            each DIASource.
+        score_struct : lsst.pipe.base.Struct
+            A struct containing two arrays: scores DIAObjects, indices of]
+            DIAObject mached to. Both arrays should be the same length as the
+            source catalog.
 
         Returns
         -------
         Indices of newly updated and created DIAObjects
         """
-        pass
+
+        used_dia_object = np.zeros(len(self.dia_objects), dtype=np.bool)
+        used_dia_source = np.zeros(len(dia_source_catalog), dtype=np.bool)
+
+        n_previous_objects = len(self.dia_objects)
+
+        updated_and_new_dia_objects = []
+
+        score_args = score_struct.scores.argsort(axis=None)
+        for score_idx in score_args:
+            if not np.isfinite(score_struct.scores[score_idx]):
+                break
+            if used_dia_object[score_idx]:
+                continue
+            used_dia_object[score_struct.indices[score_idx]] = True
+            used_dia_source[score_idx] = True
+            updated_and_new_dia_objects.append(
+                score_struct.indices[score_idx])
+
+            self.dia_objects[
+                score_struct.indices[score_idx]].append_dia_source(
+                    dia_source_catalog[score_idx])
+
+        n_new_objects = 0
+        for src_idx in np.argwhere(used_dia_source == False).flatten():
```

Comment:
numpy argwhere returns, in this case, an array of shape (N, 1) instead of (N,). This is the reason for the flatten. I've removed the flatten and written

    for (src_idx,) in np.argwhere(used_dia_source == False):

hopefully this is more clear.

Reply:
Oh, interesting. I think that's worth a very short comment.